app/data_providers/item2item.py

```diff
@@ -0,0 +1,82 @@
+import logging
+from typing import List
+
+from qdrant_client.http import AsyncApis
+from qdrant_client.http.exceptions import UnexpectedResponse
+from qdrant_client.http.models import Filter, FieldCondition, Range, MatchValue, RecommendRequest
+
+import app.config
+from app.models.corpus_item_model import CorpusItemModel
+
+
+def _build_filter(is_curated: bool = None,
+                  is_syndicated: bool = None,
+                  domain: str = None) -> Filter:
+    conditions = []
+
+    if is_syndicated is not None:
+        conditions.append(FieldCondition(
+            key='is_syndicated',
+            match=MatchValue(value=is_syndicated)
+        ))
+    if is_curated is not None:
+        conditions.append(FieldCondition(
+            key='is_curated',
+            match=MatchValue(value=is_curated)
+        ))
+    if domain is not None:
+        conditions.append(FieldCondition(
+            key='domain',
+            match=MatchValue(value=domain)
+        ))
+
+    return Filter(must=conditions)
+
+
+class Item2ItemRecommender:
+    def __init__(self):
+        host = app.config.qdrant["host"]
+        port = app.config.qdrant["port"]
+        https = app.config.qdrant["https"]
+        self.collection = app.config.qdrant["collection"]
+        self._client = AsyncApis(host=f"http{'s' if https else ''}://{host}:{port}").points_api
```

Comment:
Nit: We've started a pattern of injecting dependencies into the constructor to make code more testable and less strongly connected. Any reason not to do that here?

Reply:
I prefer high-level tests, they are less fragile after refactorings. When we want to test this part separately from the connection to qdrant, we'll extract it.